app/src/lib/stores/repositories-store.ts

```diff
@@ -531,33 +546,132 @@ export class RepositoriesStore extends BaseStore {
     return record!.lastPruneDate
   }
 
-  public async isBranchProtected(
-    gitHubRepository: GitHubRepository,
+  /**
+   * Load the branch protection information for a repository from the database
+   * and cache the results in memory
+   */
+  private async loadAndCacheBranchProtection(dbID: number) {
+    // query the database to find any protected branches
+    const branches = await this.db.protectedBranches
+      .where('repoId')
+      .equals(dbID)
+      .toArray()
+
+    const branchProtectionsFound = branches.length > 0
+    this.branchProtectionSettingsFoundCache.set(dbID, branchProtectionsFound)
+
+    // fill the retrieved records into the per-branch cache
+    for (const branch of branches) {
+      const key = getKey(dbID, branch.name)
+      this.protectionEnabledForBranchCache.set(key, true)
+    }
+
+    return branchProtectionsFound
+  }
+
+  /** Check the cache or load results from IndexedDB into cache */
+  private async findOrCacheBranchProtections(dbID: number): Promise<boolean> {
+    const branchProtectionsFound = this.branchProtectionSettingsFoundCache.get(
+      dbID
+    )
+
+    if (branchProtectionsFound === undefined) {
+      return this.loadAndCacheBranchProtection(dbID)
+    }
+
+    return branchProtectionsFound
+  }
+
+  private async isBranchProtected(
+    dbID: number,
     branchName: string
+  ): Promise<boolean> {
+    const key = getKey(dbID, branchName)
+
+    const existing = this.protectionEnabledForBranchCache.get(key)
+    if (existing === true) {
+      return existing
+    }
+
+    const result = await this.db.protectedBranches.get([dbID, branchName])
+
+    // if no row found, this means no protection is found for the branch
+    const value = result !== undefined
+
+    this.protectionEnabledForBranchCache.set(key, value)
+
+    return value
+  }
+
+  /**
+   * Check if any branch protection settings are enabled for the repository
+   * through the GitHub API.
+   *
+   * Will also check the parent repository if is defined and the fork repository
+   * does not have this branch protected.
+   */
+  public async hasBranchProtectionsConfigured(
+    gitHubRepository: GitHubRepository
   ): Promise<boolean> {
     if (gitHubRepository.dbID === null) {
       return fatalError(
         'unable to get protected branches, GitHub repository has a null dbID'
       )
     }
 
-    const key = getKey(gitHubRepository.dbID, branchName)
+    const { dbID } = gitHubRepository
+    const isBranchProtectionsFound = await this.findOrCacheBranchProtections(
+      dbID
+    )
+
+    if (isBranchProtectionsFound) {
+      return true
+    }
 
-    const existing = this.branchProtectionCache.get(key)
-    if (existing !== undefined) {
-      return existing
+    if (
+      gitHubRepository.parent !== null &&
```

Comment:
After thinking about this some more, I'm going to skip looking at the parent repository for branch protections for two reasons:

 - by checking both we're probably going to collect a lot of OSS projects that protect their default branch, but that's not relevant to people working in forks
 - when you're working in a fork, the typical workflow isn't to push to the upstream - you'd create a branch in your fork and submit that as a pull request

Reply:
yep i think that's a reasonable assumption! 🍴